samza-kv/src/main/java/org/apache/samza/storage/kv/BaseLocalStoreBackedTableDescriptor.java

```diff
@@ -36,11 +36,20 @@
  */
 abstract public class BaseLocalStoreBackedTableDescriptor<K, V, D extends BaseLocalStoreBackedTableDescriptor<K, V, D>>
     extends BaseTableDescriptor<K, V, D> {
+
+  static final public String INTERNAL_ENABLE_CHANGELOG = "internal.enable.changelog";
```

Comment:
The reason is that when we create the TableSpec, we don't know the job name and id, so the info has to be stashed for later use during planning phase.

Reply:
all the more reason to use the descriptor in the planning phase so that we can get rid of the internal placeholders for passing configurations around. Looking forward to the refactor of using descriptors in the planning phase :)

Reply:
:)